### ISSUE TITLE
[FR] Upper / Lower for multiple headings 

### ISSUE THREAD

**Issue by hpchavaz** [Benature/obsidian-text-format#83]
At present, it is possible to "upper" or "lower" a single heading.
The proposal is to extend this functionality to all headings in the current selection.

# Example
## Selected text
```
text a
# heading 1
text b
## header 2
text b
# header 3
```

## Result
```
text a
## header 1
text b
### header 2
text b
## header 3
```

# Discussion

This breaks the current use that to "upper" a simple text to a level 1 heading.

Conversely, the result to "lower" a level 1 heading would be also modified.

In fact, current behavior is not very satisfactory. A "lowered" level 1 heading is transformed into plain text, even though it would be logically coherent for it to be given a level giving it even more importance than level 1. 

This seems to be due to confusion between the semen level of the title and the number representing its level. To be clearer, a level 1 title should be considered as having a higher meaning than a level 2 title.

However, it is not possible in "Markdown" to give to headings a higher level of meaning than level 1 heading.

It is therefore proposed that the lowering of a level 1 title should have no effect.

**Comment by Benature** [comment]
Here is my understanding of this issue, please tell me if I understand something wrong.

The issue contains two request
1. Request 1: support multi-lines
2. Request 2: disable downgrade level 1 to plain text

Here's my plan:

- For request 1, the plugin detects the selection at first. If the selection is multi-line, only upgrade heading level >= 1, i.e., ignore upgrading plain text.
- For request 2, I think there is still a need for downgrading level 1 to plain text. As a walkaround to satisfy different users, I'm considering adding a toggle setting about it. Another idea is to only downgrade heading level 1 if the same command is called twice continuously (It is more troublesome, so it may not be implemented very soon).

**Comment by Benature** [comment]
Two requests are implemented in v3.0.0. The configuration is under "Markdown quicker".
